src/dft/backends/cufft/CMakeLists.txt

```diff
@@ -48,7 +48,7 @@ target_link_libraries(${LIB_OBJ} PUBLIC ONEMKL::SYCL::SYCL ${MKL_LINK_SYCL})
 set_target_properties(${LIB_OBJ} PROPERTIES
   POSITION_INDEPENDENT_CODE ON
 )
-target_link_libraries(${LIB_NAME} PUBLIC ${LIB_OBJ})
+target_link_libraries(${LIB_NAME} PUBLIC ${LIB_OBJ} onemkl_dft)
```

Comment:
It makes sense to me that we add `onemkl_dft` here but it does not seem to match with other domains. I'm not sure where else `onemkl_dft` was linked before.